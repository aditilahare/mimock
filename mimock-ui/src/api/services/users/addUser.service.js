import { post } from 'api/AxiosClient';
import Bcrypt from 'bcryptjs';

export const addNewUser = async ({ name, userName, password, userRole }) => {
	const salt = Bcrypt.genSaltSync(12);
	const hashedPassword = Bcrypt.hashSync(password, salt);

	return await post(
		'/admin/users',
		{
			name,
			userName,
			userRole,
			password: hashedPassword,
		},
		undefined,
		{ showFullPageLoader: true }
	).catch((err) => Promise.reject(err));
};
