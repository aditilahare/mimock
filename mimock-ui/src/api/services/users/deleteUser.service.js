import { remove } from 'api/AxiosClient';

export const deleteUser = async (userName) => {
	const url = '/admin/users/delete-user?userName=' + userName;

	return await remove(url)
		.then((res) => res.data)
		.catch((err) => Promise.reject(err));
};
