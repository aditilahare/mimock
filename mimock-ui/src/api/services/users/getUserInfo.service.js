import { get } from 'api/AxiosClient';

export const getUserInfo = async (userId) => {
	const url = `/admin/users/user-info?userId=${userId}`;

	return await get(url, { showFullPageLoader: true })
		.then((res) => res.data.userInfo)
		.catch((err) => Promise.reject(err));
};
