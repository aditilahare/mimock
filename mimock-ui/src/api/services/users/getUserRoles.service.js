import { get } from 'api/AxiosClient';

export const getUserRoles = async () => {
	return await get('/static-records/user-roles')
		.then((res) => res.data)
		.catch((err) => err);
};
