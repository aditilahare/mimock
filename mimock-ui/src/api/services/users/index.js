import { getAllUsers } from './getUsers.service';
import { deleteUser } from './deleteUser.service';

export { getAllUsers, deleteUser };
