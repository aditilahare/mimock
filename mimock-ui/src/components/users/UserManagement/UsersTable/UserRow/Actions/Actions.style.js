import tw from 'tailwind-styled-components';

export const ActionsWrapper = tw.div`
    w-1/4
`;

export const Options = tw.div`
    text-center
    text-2xl
    text-gray-600
    cursor-pointer
    flex
    justify-center
    items-center
    align-middle
    gap-4
    w-full
`;
