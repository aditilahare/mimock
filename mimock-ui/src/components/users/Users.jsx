import React from 'react';
import UserManagement from './UserManagement';

export default function Users() {
	return <UserManagement />;
}
