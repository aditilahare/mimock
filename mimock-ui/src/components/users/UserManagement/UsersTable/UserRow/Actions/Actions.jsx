import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useRecoilState } from 'recoil';
import { FaEdit, FaTrash } from 'react-icons/fa';
import Tooltip from '@mui/material/Tooltip';
import IconButton from '@mui/material/IconButton';
import { deleteUser } from 'services/users';
import { ConfirmationModal } from 'components/common/Modals';
import useNotification from 'hooks/useNotification';
import deletionModalAtom from 'atoms/deletionModalAtom';
import {
	notificationTypes,
	notificationPositions,
} from 'constants/notificationConstants';
import { ActionsWrapper, Options } from './Actions.style';

function Actions({ userName }) {
	const [showDeletionModal, setShowDeletionModal] = useState(false);
	const [deletingUser, setDeletingUser] = useState(false);
	const [, setRefreshUserList] = useRecoilState(deletionModalAtom);

	const deletionConfirmationMessage = `Are you sure you want to delete user "${userName}" ?`;
	const deletingMessage = 'Deleting user. Please wait...';

	const options = [
		{
			name: 'edit',
			icon: <FaEdit />,
			tooltip: 'Edit User',
			onClick: () => {},
		},
		{
			name: 'delete',
			icon: <FaTrash />,
			tooltip: 'Delete User',
			onClick: () => {
				setShowDeletionModal(true);
				setRefreshUserList(true);
			},
		},
	];

	const handleDeleteUser = async () => {
		setDeletingUser(true);

		await deleteUser(userName)
			.then(() => {
				useNotification({
					type: notificationTypes.NOTIFICATION_TYPE_SUCCESS,
					title: 'Deletion successful',
					message: `User - ${userName} has been deleted`,
					position: notificationPositions.NOTIFICATION_POSITION_BOTTOM_RIGHT,
					animationIn: 'animate__slideInRight',
					animationOut: 'animate__slideOutRight',
				});

				setRefreshUserList(false);
				setDeletingUser(false);
				setShowDeletionModal(false);
			})
			.catch(() => {
				useNotification({
					type: notificationTypes.NOTIFICATION_TYPE_ERROR,
					title: `Failed to delete user - ${userName}`,
					message: 'Please try again',
					animationIn: 'animate__bounceIn',
					animationOut: 'animate__bounceOut',
				});

				setRefreshUserList(false);
				setDeletingUser(false);
				setShowDeletionModal(false);
			});
	};

	return (
		<ActionsWrapper data-testid={`user-action-${userName}`}>
			<If condition={showDeletionModal}>
				<ConfirmationModal
					message={deletionConfirmationMessage}
					cancelButtonLabel='Do not delete user'
					confirmButtonLabel='Delete user'
					loading={deletingUser}
					loadingMessage={deletingMessage}
					onConfirm={async () => {
						await handleDeleteUser();
					}}
					onCancel={() => {
						setShowDeletionModal(false);
					}}
				/>
			</If>
			<Options>
				<For each='option' of={options}>
					<Tooltip key={option.name} title={option.tooltip} arrow>
						<IconButton
							key={`${option.name}-${userName}`}
							data-testid={`${option.name}-${userName}`}
							onClick={() => option.onClick(userName)}
						>
							{option.icon}
						</IconButton>
					</Tooltip>
				</For>
			</Options>
		</ActionsWrapper>
	);
}

Actions.propTypes = {
	userName: PropTypes.string.isRequired,
};

export default Actions;
