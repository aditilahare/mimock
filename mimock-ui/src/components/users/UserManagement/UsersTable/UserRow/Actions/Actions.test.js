import React from 'react';
import { render, act, fireEvent } from '@testing-library/react';
import { deleteUser } from 'services/users';
import Actions from './Actions';

const actualNotification = jest.requireActual('hooks/useNotification');

let mockedRecoilFn;
jest.mock('recoil', () => {
	mockedRecoilFn = jest.fn();
	return {
		atom: jest.fn(),
		useRecoilState: jest.fn(() => [true, mockedRecoilFn]),
	};
});
jest.mock('react-notifications-component', () => {
	const Store = {
		addNotification: jest.fn(),
	};
	return { Store };
});

const mockedNavigation = jest.fn(() => {});
jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useNavigate: () => mockedNavigation,
}));
jest.mock('services/users/deleteUser.service');

describe('Actions', () => {
	it('should render user actions', async () => {
		let tree;
		await act(async () => {
			tree = await render(
				<Actions
					userName='luke1'
					userId='d749184e-b60d-4ac0-b459-e3b9cc5710d1'
				/>
			);
		});

		const { container, getByTestId } = tree;

		expect(getByTestId('user-action-luke1')).toBeInTheDocument();

		expect(getByTestId('edit-luke1')).toBeInTheDocument();
		expect(getByTestId('delete-luke1')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	it('should show confirmation modal on clicking delete', async () => {
		let tree;
		await act(async () => {
			tree = await render(
				<Actions
					userName='luke1'
					userId='d749184e-b60d-4ac0-b459-e3b9cc5710d1'
				/>
			);
		});

		const { getByTestId } = tree;

		expect(getByTestId('user-action-luke1')).toBeInTheDocument();

		expect(getByTestId('edit-luke1')).toBeInTheDocument();

		await act(async () => {
			await fireEvent.click(getByTestId('delete-luke1'));
		});

		expect(mockedRecoilFn).toHaveBeenCalledTimes(1);
		expect(mockedRecoilFn).toHaveBeenCalledWith(true);

		expect(getByTestId('confirmation-modal')).toBeInTheDocument();

		expect(document.body).toMatchSnapshot();
	});

	it('should navigate to edit user page on clicking edit button', async () => {
		let tree;
		await act(async () => {
			tree = await render(
				<Actions
					userName='luke1'
					userId='d749184e-b60d-4ac0-b459-e3b9cc5710d1'
				/>
			);
		});

		const { getByTestId } = tree;

		expect(getByTestId('user-action-luke1')).toBeInTheDocument();

		expect(getByTestId('delete-luke1')).toBeInTheDocument();

		await act(async () => {
			await fireEvent.click(getByTestId('edit-luke1'));
		});

		expect(mockedNavigation).toHaveBeenCalledTimes(1);
		expect(mockedNavigation).toHaveBeenCalledWith(
			'/mimock-ui/admin/users/edit?userId=d749184e-b60d-4ac0-b459-e3b9cc5710d1'
		);
	});

	it('should close confirmation modal on clicking cancel', async () => {
		const notificationSpy = jest.spyOn(actualNotification, 'default');

		let tree;
		await act(async () => {
			tree = await render(
				<Actions
					userName='luke1'
					userId='d749184e-b60d-4ac0-b459-e3b9cc5710d1'
				/>
			);
		});

		const { getByTestId, queryByTestId } = tree;

		expect(getByTestId('edit-luke1')).toBeInTheDocument();

		act(() => {
			const deleteUserBtn = getByTestId('delete-luke1');
			fireEvent.click(deleteUserBtn);
		});

		expect(getByTestId('confirmation-modal')).toBeInTheDocument();
		expect(getByTestId('confirmation-modal-message')).toBeInTheDocument();
		expect(getByTestId('confirmation-modal-message').textContent).toBe(
			'Are you sure you want to delete user "luke1" ?'
		);

		await act(async () => {
			const confirmDeleteBtn = getByTestId('confirmation-modal-cancel-btn');
			await fireEvent.click(confirmDeleteBtn);
		});

		expect(deleteUser).not.toHaveBeenCalledTimes(1);
		expect(notificationSpy).not.toHaveBeenCalled();

		expect(queryByTestId('confirmation-modal')).not.toBeInTheDocument();

		expect(document.body).toMatchSnapshot();
	});

	it('should invoke delete user service on confirming user deletion', async () => {
		deleteUser.mockResolvedValue({
			id: 'luke1',
			message: 'User deleted successfully',
		});
		const notificationSpy = jest.spyOn(actualNotification, 'default');

		let tree;
		await act(async () => {
			tree = await render(
				<Actions
					userName='luke1'
					userId='d749184e-b60d-4ac0-b459-e3b9cc5710d1'
				/>
			);
		});

		const { getByTestId } = tree;

		expect(getByTestId('edit-luke1')).toBeInTheDocument();

		act(() => {
			const deleteUserBtn = getByTestId('delete-luke1');
			fireEvent.click(deleteUserBtn);
		});

		expect(getByTestId('confirmation-modal')).toBeInTheDocument();
		expect(getByTestId('confirmation-modal-message')).toBeInTheDocument();
		expect(getByTestId('confirmation-modal-message').textContent).toBe(
			'Are you sure you want to delete user "luke1" ?'
		);

		await act(async () => {
			const confirmDeleteBtn = getByTestId('confirmation-modal-confirm-btn');
			await fireEvent.click(confirmDeleteBtn);
		});

		expect(deleteUser).toHaveBeenCalledTimes(1);
		expect(deleteUser).toHaveBeenCalledWith('luke1');

		expect(notificationSpy).toHaveBeenCalledTimes(1);
		expect(notificationSpy).toHaveBeenCalledWith({
			type: 'success',
			title: 'Deletion successful',
			message: 'User - luke1 has been deleted',
			position: 'bottom-right',
			animationIn: 'animate__slideInRight',
			animationOut: 'animate__slideOutRight',
		});

		expect(document.body).toMatchSnapshot();
	});

	it('should show error notification when user deletion fails', async () => {
		deleteUser.mockRejectedValue(new Error('Deletion failed!'));
		const notificationSpy = jest.spyOn(actualNotification, 'default');

		let tree;
		await act(async () => {
			tree = await render(
				<Actions
					userName='luke1'
					userId='d749184e-b60d-4ac0-b459-e3b9cc5710d1'
				/>
			);
		});

		const { getByTestId } = tree;

		expect(getByTestId('edit-luke1')).toBeInTheDocument();

		act(() => {
			const deleteUserBtn = getByTestId('delete-luke1');
			fireEvent.click(deleteUserBtn);
		});

		expect(getByTestId('confirmation-modal')).toBeInTheDocument();
		expect(getByTestId('confirmation-modal-message')).toBeInTheDocument();
		expect(getByTestId('confirmation-modal-message').textContent).toBe(
			'Are you sure you want to delete user "luke1" ?'
		);

		await act(async () => {
			const confirmDeleteBtn = getByTestId('confirmation-modal-confirm-btn');
			await fireEvent.click(confirmDeleteBtn);
		});

		expect(deleteUser).toHaveBeenCalledTimes(1);
		expect(deleteUser).toHaveBeenCalledWith('luke1');

		expect(notificationSpy).toHaveBeenCalledTimes(1);
		expect(notificationSpy).toHaveBeenCalledWith({
			type: 'danger',
			title: 'Failed to delete user - luke1',
			message: 'Please try again',
			animationIn: 'animate__bounceIn',
			animationOut: 'animate__bounceOut',
		});

		expect(document.body).toMatchSnapshot();
	});
});
