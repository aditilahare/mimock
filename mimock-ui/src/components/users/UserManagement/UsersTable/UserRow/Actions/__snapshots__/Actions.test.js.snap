// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions should close confirmation modal on clicking cancel 1`] = `
<body>
  <div>
    <div
      class="w-1/4"
      data-testid="user-action-luke1"
    >
      <div
        class="text-center text-2xl text-gray-600 cursor-pointer flex justify-center items-center align-middle gap-4 w-full"
      >
        <button
          aria-label="Edit User"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="edit-luke1"
          tabindex="0"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="Delete User"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="delete-luke1"
          tabindex="0"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`Actions should invoke delete user service on confirming user deletion 1`] = `
<body>
  <div>
    <div
      class="w-1/4"
      data-testid="user-action-luke1"
    >
      <div
        class="text-center text-2xl text-gray-600 cursor-pointer flex justify-center items-center align-middle gap-4 w-full"
      >
        <button
          aria-label="Edit User"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="edit-luke1"
          tabindex="0"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="Delete User"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="delete-luke1"
          tabindex="0"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`Actions should render user actions 1`] = `
<div>
  <div
    class="w-1/4"
    data-testid="user-action-luke1"
  >
    <div
      class="text-center text-2xl text-gray-600 cursor-pointer flex justify-center items-center align-middle gap-4 w-full"
    >
      <button
        aria-label="Edit User"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        data-testid="edit-luke1"
        tabindex="0"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="Delete User"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        data-testid="delete-luke1"
        tabindex="0"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Actions should show confirmation modal on clicking delete 1`] = `
<body>
  <div>
    <div
      class="w-1/4"
      data-testid="user-action-luke1"
    >
      <div
        class="text-center text-2xl text-gray-600 cursor-pointer flex justify-center items-center align-middle gap-4 w-full"
      >
        <button
          aria-label="Edit User"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="edit-luke1"
          tabindex="0"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="Delete User"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="delete-luke1"
          tabindex="0"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div>
    <div
      class="w-screen h-screen flex justify-center items-center bg-slate-500/[0.8] align-middle fixed left-0 right-0 top-0 bottom-0"
      data-testid="confirmation-modal"
    >
      <div
        class="xl:w-1/3 lg:w-1/3 md:w-1/2 rounded-md shadow-lg bg-white p-6"
      >
        <div
          class="text-center font-sans font-semibold text-xl my-4 text-gray-700"
          data-testid="confirmation-modal-message"
        >
          Are you sure you want to delete user "luke1" ?
        </div>
        <div
          class="w-1/2 text-center font-sans font-semibold text-lg mx-auto mt-8 cursor-pointer transition-all duration-100 hover:scale-95 text-gray-500 hover:text-gray-900"
          data-testid="confirmation-modal-cancel-btn"
        >
          Do not delete user
        </div>
        <div
          class="w-1/2 text-center font-sans font-semibold text-lg mx-auto mt-8 cursor-pointer transition-all duration-100 hover:scale-95 bg-teal-500 text-white rounded-full shadow-md p-4 mb-4 hover:bg-teal-600"
          data-testid="confirmation-modal-confirm-btn"
        >
          Delete user
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Actions should show error notification when user deletion fails 1`] = `
<body>
  <div>
    <div
      class="w-1/4"
      data-testid="user-action-luke1"
    >
      <div
        class="text-center text-2xl text-gray-600 cursor-pointer flex justify-center items-center align-middle gap-4 w-full"
      >
        <button
          aria-label="Edit User"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="edit-luke1"
          tabindex="0"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="Delete User"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="delete-luke1"
          tabindex="0"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`Actions should trigger edit user action on clicking edit button 1`] = `
<body>
  <div>
    <div
      class="w-1/4"
      data-testid="user-action-luke1"
    >
      <div
        class="text-center text-2xl text-gray-600 cursor-pointer flex justify-center items-center align-middle gap-4 w-full"
      >
        <button
          aria-label="Edit User"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="edit-luke1"
          tabindex="0"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="Delete User"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="delete-luke1"
          tabindex="0"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;
